lighthouse-core/gather/gatherers/animated-elements.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('./gatherer.js');
+const TraceProcessor = require('../../lib/tracehouse/trace-processor.js');
+const pageFunctions = require('../../lib/page-functions.js');
+
+/* eslint-env browser, node */
+
+/**
+ * @this {HTMLElement}
+ * @param {number} nodeId
+ * @return {LH.Artifacts.AnimatedElement | undefined}
+ */
+/* istanbul ignore next */
+function getNodeData(nodeId) {
+  const elem = this.nodeType === document.ELEMENT_NODE ? this : this.parentElement; // eslint-disable-line no-undef
+  let animatedElement;
+  if (elem) {
+    animatedElement = {
+      nodeId,
+      // @ts-ignore - put into scope via stringification
+      devtoolsNodePath: getNodePath(elem), // eslint-disable-line no-undef
+      // @ts-ignore - put into scope via stringification
+      selector: getNodeSelector(elem), // eslint-disable-line no-undef
+      // @ts-ignore - put into scope via stringification
+      nodeLabel: getNodeLabel(elem), // eslint-disable-line no-undef
+      // @ts-ignore - put into scope via stringification
+      snippet: getOuterHTMLSnippet(elem), // eslint-disable-line no-undef
+    };
+  }
+  return animatedElement;
+}
+
+class AnimatedElements extends Gatherer {
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @param {LH.Gatherer.LoadData} loadData
+   * @return {Promise<LH.Artifacts['AnimatedElements']>}
+   * @override
+   */
+  async afterPass(passContext, loadData) {
+    const driver = passContext.driver;
+    if (!loadData.trace) {
+      throw new Error('Trace is missing!');
+    }
+
+    const {mainThreadEvents} = TraceProcessor.computeTraceOfTab(loadData.trace);
```

Comment:
definitely unfortunate to try to run computeTraceOfTab twice :/

how busy is this `Animation` event? is it worth just surfacing *all* and linking up to main-thread in the audit?

Reply:
The reason I created this gatherer was so we could resolve the node ids in the animation.

I can probably move the audit code which fetches the failure reasons here and report them alongside each node so we don't need to compute the trace of tab twice.

Reply:
Moving anything from audit -> gatherer won't help since trace of tab is run by a ton of other audits, it's free on the audit side. Doing in here in the gatherer without computed artifacts cache is the expensive part.

I wasn't trying to suggest we don't need the gatherer at all either :)

I was asking if there are *tons* of `Animation` events off the main-thread such that we need to filter them here. If we don't expect many animation events off the main thread, then we can just skip the filtering step here and return all animation events to the audit (which can then filter out the main-thread ones)

Reply:
This gatherer is not supposed to be a filter for animation events, it is supposed to fetch animated node info (label, snippet, path,   selector) using so that the nodes affected by each animation can be listed in the report similar to `largest-contentful-paint-element`.

Is there some way to fetch all animated nodes without computing trace of tab here?